pkg/controller/operators/catalog/operator.go

```diff
@@ -2074,20 +2092,24 @@ func validateV1Beta1CRDCompatibility(dynamicClient dynamic.Interface, oldCRD *ap
 	return nil
 }
 
-func validateExistingCRs(dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, newCRD *apiextensions.CustomResourceDefinition) error {
-	// make dynamic client
+// validateExistingCRs will list all CRs from a given gvr then validate each with the new validation
+func validateExistingCRs(dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, crdValidation *apiextensions.CustomResourceValidation) error {
 	crList, err := dynamicClient.Resource(gvr).List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		return fmt.Errorf("error listing resources in GroupVersionResource %#v: %s", gvr, err)
 	}
+
+	// create validator from given crdValidation
+	validator, _, err := validation.NewSchemaValidator(crdValidation)
+	if err != nil {
+		return fmt.Errorf("error creating validator for schema %#v: %s", crdValidation, err)
+	}
+
+	// validate each CR against this version schema
 	for _, cr := range crList.Items {
-		validator, _, err := validation.NewSchemaValidator(newCRD.Spec.Validation)
-		if err != nil {
-			return fmt.Errorf("error creating validator for schema %#v: %s", newCRD.Spec.Validation, err)
-		}
 		err = validation.ValidateCustomResource(field.NewPath(""), cr.UnstructuredContent(), validator).ToAggregate()
 		if err != nil {
-			return fmt.Errorf("error validating custom resource against new schema for %s %s/%s: %v", newCRD.Spec.Names.Kind, cr.GetNamespace(), cr.GetName(), err)
+			return fmt.Errorf("error validating custom resource against new schema for version %s of %s %s/%s: %v", cr.GetAPIVersion(), cr.GroupVersionKind().Kind, cr.GetNamespace(), cr.GetName(), err)
```

Comment:
If it's cluster-scoped, namespace will be empty. Let's format the fully qualified name either as %s or %s/%s based on cluster vs. namespace scoped.